### PR TITLE
Extend dkg scenario tests

### PR DESCRIPTION
**Pull request by wrinkledeth**
### PR Notes 

- Fixed bug in controller emitGroupEvent()
- Implemented 9 new scenario tests (98% test coverage reached)
- Uploaded detailed notes describing all test scenarios.

### New scenario tests
  - test33333NodeRegister()
  - test33NodeRegister()
  - test53NodeQuit()
  - test5NodeQuit()
  - test63Group0NodeQuit()
  - test63Group1NodeQuit()
  - test66666NodeRegister()
  - test66NodeRegister()
  - testNodeSetup()

### CODE REVIEW

File: contracts/test/ExtendedDKGScenarioTests.t.sol [hunk 1]
```diff
@@ -0,0 +1,770 @@
...
+        // Final State: group_0 (1,2,3), group_1 (4,5,6), group_2 (7)
+        // Group 2 is not yet functional, but it is created. Group 0 and 1 are functional.
+
+        printGroupInfo(0);
+        printGroupInfo(1);
+        printGroupInfo(2);
+    }
+
+    // Ideal number of groups of size 3 -> new nodeRegister()
+    // [3,3,3,3,3] -> nodeRegister -> [3,3,3,3,4]
```
Comment (kafeikui) [inline contracts/test/ExtendedDKGScenarioTests.t.sol:484]: [4,3,3,3,3] I guess？

File: contracts/test/ExtendedDKGScenarioTests.t.sol [hunk 2]
```diff
@@ -0,0 +1,770 @@
...
+        assertEq(checkIsStrictlyMajorityConsensusReached(3), true);
+        assertEq(checkIsStrictlyMajorityConsensusReached(4), true);
+
+        // New node calls node register: It gets added to group_0, new event is emitted
+        registerIndex(16);
+        assertEq(controller.getGroup(0).size, 4);
+        assertEq(checkIsStrictlyMajorityConsensusReached(0), false);
+        assertEq(controller.getGroup(0).epoch, 2);
+
+        // Final State: [3,3,3,3,4]
```
Comment (kafeikui) [inline contracts/test/ExtendedDKGScenarioTests.t.sol:567]: The same above

**Comment by kafeikui** [comment]
LGTM! Good job on construction of the cases' initial state!

**Comment by wrinkledeth** [comment]
Closing this PR. 

Multiple large changes were needed to fix merge conflicts, so a new PR was opened here: https://github.com/ARPA-Network/BLS-TSS-Network/pull/24

**Pull request closed without merging**